provider/plural/client.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plural
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/pluralsh/gqlclient"
+	"github.com/pluralsh/gqlclient/pkg/utils"
+)
+
+type authedTransport struct {
+	key     string
+	wrapped http.RoundTripper
+}
+
+func (t *authedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("Authorization", "Bearer "+t.key)
+	return t.wrapped.RoundTrip(req)
+}
+
+type Client interface {
+	DnsRecords() ([]*DnsRecord, error)
+	CreateRecord(record *DnsRecord) (*DnsRecord, error)
+	DeleteRecord(name, ttype string) error
+}
+
+type Config struct {
+	Token    string
+	Endpoint string
+	Cluster  string
+	Provider string
+}
+
+type client struct {
+	ctx          context.Context
+	pluralClient *gqlclient.Client
+	config       *Config
+}
+
+type DnsRecord struct {
+	Type    string
+	Name    string
+	Records []string
+}
+
+func NewClient(conf *Config) Client {
+	base := conf.BaseUrl()
+	httpClient := http.Client{
+		Transport: &authedTransport{
+			key:     conf.Token,
+			wrapped: http.DefaultTransport,
+		},
+	}
+	endpoint := base + "/gql"
+	return &client{
+		ctx:          context.Background(),
+		pluralClient: gqlclient.NewClient(&httpClient, endpoint),
+		config:       conf,
+	}
+}
+
+func (c *Config) BaseUrl() string {
+	host := "https://app.plural.sh"
+	if c.Endpoint != "" {
+		host = fmt.Sprintf("https://%s", c.Endpoint)
```

Comment:
better make sure that it's actually an URL, so url.Parse and handle error

Reply:
done